### ISSUE TITLE
CRAN release looming

### ISSUE THREAD

**Issue by jarioksa** [vegandevs/vegan#509]
Due to changes in R 4.2.0 we **must** have a new release of **vegan** (see resolved issues #507,  #447). The deadline CRAN gives us is April 23. Let's keep to it.

This would be release 2.6.0. The 2.5-branch is diverged too much, and we may just close the gap by having 2.6.0 (or 2.6.1 to keep 2.6.0 as the pre-release github version).

If there are any other things you want to get in the **vegan**, now you know the timeline.

**Comment by jarioksa** [comment]
I checked the reverse dependences against CRAN release `vegan 2.5-7`:

The deprecation of `adonis` in favour of `adonis2` will trigger warnings in examples of the following packages:

**betaper** **dispRity** **GUniFrac** **microeco** **smartsnp**

I have contacted the authors of these packages. I will also delay the deprecation of `adonis` for this minor release so that these package authors can adapt to the deprecation (but print a `message` of the future deprecation).

Several packages failed their tests, but for reasons unrelated to **vegan**. The only packages had different test results in 2.5-7 and 2.6-0 are those listed above.

Reverse dependences were checked in 170 packages and 609 packages had to be installed to satisfy the dependences of these tested packages. I'm happy that I have a fast computer with several cores for parallel testing.

**Comment by jarioksa** [comment]
First release candidate (2.6-1) was rejected, because it broke CRAN package **microbial**. That was not our fault, but the error was in BioC package **phyloseq** which assumed **vegan** pre-2.0-10 (Dec 2013) data structure, if `stats::uniroot` issued a warning in `vegan::fisherfit`.  A quick trick was to go back to the old bracketing strategy in `stats::uniroot` which was changed for other reasons in commit 69b59dc0b4. Now we will get an error with all-singleton communities, such as a "community" made of one individual: try `fisherfit(1)` or `fisher.alpha(1)`. The only sustainable solution is that **phyloseq** is fixed so that it does not assume a pre-2014 result structure when `fisher.alpha` issues a warning (so that warning leads to an error in **phyloseq**). I have published an issue and provided a pull request in **phyloseq**, but I do not expect that these lead to any changes very soon. However, the change made for 2.6-2 allowed acceptance of **vegan** in CRAN, since the other cases with new warnings were not used in any dependent package.
